modules/dashboard-ui/dashboard-ui-core/src/main/java/org/jboss/dashboard/ui/taglib/PropertyReadTag.java

```diff
@@ -128,19 +128,18 @@ protected Object getPropertyValue() {
             return null;
         }
         try {
-            Object val = JXPathContext.newContext(subjectOfTheGetter).getValue(property);
-            if (localize != null && localize.booleanValue() && val instanceof Map) {
-                return StringEscapeUtils.ESCAPE_ECMASCRIPT.translate(
-                        StringEscapeUtils.ESCAPE_HTML4.translate((String) (localize((Map) val))));
-            }
-            return StringEscapeUtils.ESCAPE_ECMASCRIPT.translate(StringEscapeUtils.ESCAPE_HTML4.translate((String)val));
-        } catch (Exception e) {
+            Object value = JXPathContext.newContext(subjectOfTheGetter).getValue(property);
+            return formatValue(LocaleManager.lookup(), value, localize);
+        }
+        catch (Exception e) {
             log.warn("Error accessing property " + property + " in " + object + "." + e);
             return null;
         }
     }
 
-    protected Object localize(Map map) {
-        return LocalizeTag.getLocalizedValue(map, LocaleManager.currentLang(), true);
+    public static String formatValue(LocaleManager localeManager, Object value, Boolean localize) {
+        boolean requiresL18n = localize != null && localize && value instanceof Map;
```

Comment:
This method could need a couple of refactorings: 
- requiresI18n (not L18n)
- Boolean.TRUE.equals(localize) instead of localize !=null && localize
- Method overloads instead of instanceof checks